### PR TITLE
feat(protocol-designer): wire up the settings page

### PR DESCRIPTION
**Pull request by jerader**
closes AUTH-795 AUTH-794

# Overview

This PR adds the settings page route to react-router-dom and wires up the page. Now, you can access the hints, analytics toggle, and dev ffs on that page

Screenshot 2024-09-17 at 12 16 20

also here are the designs: https://www.figma.com/design/WbkiUyU8VhtKz0JSuIFA45/Feature%3A-Protocol-Designer-Phase-1?node-id=9074-118563&node-type=instance&t=NXK49JTl7Fsni05A-0

## Test Plan and Hands on Testing

Test accessing the settings icon from the landing page, the overview, and deck setup. If you click on the settings button again while on the settings page, it should return back to the previous page you were on

## Changelog

- delete the feature flag modal and bouncing button 😬 and add the settings page which is home to the: dev ffs, hints, pd version, and analytics toggle
- wire up the settings icon to the navigation bar
- add test coverage

## Risk assessment

low

### CODE REVIEW

File: protocol-designer/src/atoms/SettingsIcon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      width="2rem"
+      height="2rem"
+      justifyContent={JUSTIFY_CENTER}
+    >
+      <Btn
+        onClick={handleNavigate}
+        css={BUTTON_LINK_STYLE}
+        data-testid="SettingsIconButton"
+      >
+        <Icon size="1rem" name="settings" />
```
Comment (ncdiehl11) [inline protocol-designer/src/atoms/SettingsIcon/index.tsx]:
```suggestion
        <Flex justifyContent={JUSTIFY_CENTER}>
          <Icon size="1rem" name="settings" />
        </Flex>
```

**Review by ncdiehl11 — APPROVED** [review]
just 1 style comment 🤩

File: protocol-designer/src/atoms/SettingsIcon/index.tsx
Comment (koji) [inline protocol-designer/src/atoms/SettingsIcon/index.tsx:1]: this is using components in atoms so `protocol-designer/src/molecules/SettingsIcon/index.tsx` would be better.

File: protocol-designer/src/pages/Settings/index.tsx [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+  const prereleaseFlagRows = allFlags.map(toFlagRow)
+
+  return (
+    <Flex
+      backgroundColor={COLORS.grey10}
+      width="100%"
+      minHeight="calc(100vh - 56px)"
+      height="100%"
+      padding="80px 272px"
```
Comment (koji) [inline protocol-designer/src/pages/Settings/index.tsx]:
```suggestion
      padding={` ${SPACING.spacing80} 17rem`}
```

File: protocol-designer/src/pages/Settings/index.tsx [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+                  {t('shared:reset_hints_and_tips')}
+                </StyledText>
+              </Flex>
+            </Flex>
+            <Btn
+              disabled={!canClearHintDismissals}
+              textDecoration={
+                canClearHintDismissals
+                  ? TYPOGRAPHY.textDecorationUnderline
+                  : FLEX_AUTO
```
Comment (koji) [inline protocol-designer/src/pages/Settings/index.tsx]: what is this?
maybe `none`?

File: protocol-designer/src/pages/Settings/index.tsx [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+              css={
+                Boolean(hasOptedIn)
+                  ? TOGGLE_ENABLED_STYLES
+                  : TOGGLE_DISABLED_STYLES
+              }
+              onClick={() => dispatch(_toggleOptedIn())}
+            >
+              <Icon
+                name={hasOptedIn ? 'ot-toggle-input-on' : 'ot-toggle-input-off'}
+                height="1rem"
```
Comment (koji) [inline protocol-designer/src/pages/Settings/index.tsx:207]: maybe `size`?

File: protocol-designer/src/pages/Settings/index.tsx [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+              </StyledText>
+              <Flex color={COLORS.grey60}>
+                <StyledText desktopStyle="bodyDefaultRegular">
+                  {t('shared:we_are_improving')}
+                </StyledText>
+              </Flex>
+            </Flex>
+            <Btn
+              role="switch"
+              data-testid="analyticsToggle"
```
Comment (koji) [inline protocol-designer/src/pages/Settings/index.tsx:196]: probably we should add `aria-checked`

```suggestion
              ariaChecked={Boolean(hasOptedIn)}
              data-testid="analyticsToggle"
```

File: protocol-designer/src/atoms/SettingsIcon/index.tsx
Comment (jerader) [reply protocol-designer/src/atoms/SettingsIcon/index.tsx:1]: thanks Koji, i always get confused deciding between molecules and atoms

File: protocol-designer/src/atoms/SettingsIcon/index.tsx (hunk 1 above)
Comment (jerader) [reply protocol-designer/src/atoms/SettingsIcon/index.tsx]: oops good catch!

**Pull request merged**